### ISSUE TITLE
Using of cppan packages without cmake

### ISSUE THREAD

**Issue by zdenop** [cppan/cppan#53]
Is there possibility to "reasonably" use cppan without cmake?

After some years ;-) I tried to build qt-box-editor on windows with vc 2017 (in past I used mingw). Because it is Qt based project I would prefare to use qmake instead of cmake. I need there leptonica and tesseract library so I build that libraries from source with cppan.
At the moment I am able to build this project, but after integrating results from cppan build I fihished with these binaries:
```
Qt5Core.dll
Qt5Gui.dll
Qt5Network.dll
Qt5Svg.dll
Qt5Widgets.dll
leptonica-1.76.0.dll
pvt.cppan.demo.danbloomberg.leptonica-1.76.0.dll
pvt.cppan.demo.jpeg-9.2.0.dll
pvt.cppan.demo.madler.zlib-1.2.11.dll
pvt.cppan.demo.openjpeg.openjp2-2.3.0.dll
pvt.cppan.demo.png-1.6.34.dll
pvt.cppan.demo.tiff-4.0.9.dll
pvt.cppan.demo.webmproject.webp-1.0.0.dll
pvt.cppan.demo.webp-0.6.1.dll
pvt.cppan.demo.xz_utils.lzma-5.2.3.dll
qt-box-editor-1.13dev.exe
tesseract305.dll
```

It seems that  webp-0.6.1 and webp-1.0.0 are need at the same time. and I need to have leptonica-1.76.0.dll and pvt.cppan.demo.danbloomberg.leptonica-1.76.0.dll :-( I guess I can optimize this by using local leptonica for building tesseract instead of cppan leptonica. 

But maybe you can improve doc with suggestions how to use cppan libraries (especially is there are interdependence like tesseract use leptonica) without cmake.

**Comment by egorpugin** [comment]
Cppan has basic qt libs. You can try to use them. (I use them in small projects.)
https://cppan.org/projects/pvt.cppan.demo.qtproject

**Comment by Telokis** [comment]
@egorpugin How is Cppan2 going? Anything new? Will it be released soon? Can we have a glimpse of it?

**Comment by Telokis** [comment]
Thank you very much for the (quick) answer!
Do you plan on renaming the project to Software Network? (I like `cppan` more)

**Comment by egorpugin** [comment]
Yes, because it will handle other languages.
Initial name is not very suitable anymore.

**Comment by Telokis** [comment]
Oh ok! Seems like a big thing, then!

(Not really related but I have kind of a support question, I'm trying to convert a CMake project to cppan and am getting a `error: no such file or directory` error when I do `cppan --build .` on this repository : https://gitlab.com/Telokis/Rayon using this simple `cppan.yml` file:
```yml
common_settings:
    c++: 14

files:
    - Source/.*
    - Include/.*

dependencies:
    - pvt.cppan.demo.boost.program_options: 1.63.0
    - pvt.cppan.demo.boost.log: 1.63.0
    - pvt.cppan.demo.open_source_parsers.jsoncpp: 1.8.0
    - pvt.cppan.demo.glm: 0.9.8
```
Do you see anything wrong? Would you like me to create another issue instead? Thanks anyway)

**Comment by egorpugin** [comment]
Not sure. Works for me, except build process itself - some issues with that json lib.

**Comment by Telokis** [comment]
Hmm... Maybe an issue because I'm on Windows and/or use VS 2017?

**Comment by Telokis** [comment]
I changed my spec file to

```yml
common_settings:
    c++: 14

files:
    - Source/.*
    - Include/.*
    - Dependencies/json/Json.cpp
    - Dependencies/json/Json.h
    - Dependencies/json/Json-forwards.h

include_directories:
    private:
        - Include/
        - Dependencies/json
        - Dependencies/glm/include

dependencies:
    - pvt.cppan.demo.boost.program_options: 1.63.0
    - pvt.cppan.demo.boost.log: 1.63.0
```

It works perfectly fine on Linux but doesn't work on Windows. It may be an issue with VS 2017 but I'm not sure since it only writes "no such file or directory" with no details whatsoever.

**Comment by egorpugin** [comment]
I tried on windows.
Maybe you have spaces/unicode in path?

**Comment by egorpugin** [comment]
> Do you plan on making a big "What changed" page once SW is ready?

Nothing big except describing build directly in C++.
Other build aspects are same since make, I guess.

> When will it be open for beta?

SW is already working (or should) in server-less mode. You could try to build from sources or I can send binary.

> Will it be backward-compatible with cppan.yml v1 files? 

cppan1 will be available as frontend, if you don't want to code rules in C++.
But it won't be possible to make cmake insertions.

For more info see this thread: https://github.com/cppan/cppan/issues/58

**Comment by egorpugin** [comment]
@biocomp @Telokis 

FYI:
I've started initial SW web app on the network: https://software-network.org/
It is now serving all sw requests instead of trying to connect to local server (it was so during my development and bootstrapping).
You can try to build/dl clients, play with them.
Very basic things like `sw build sw.cpp` should work.
Feel free to open issues with bugs/requests on the https://github.com/SoftwareNetwork/sw

Hardening, optimizations and bug hunting will be main priority in the nearest future.

**Comment by egorpugin** [comment]
Proceed to SW, please.
https://github.com/SoftwareNetwork/sw
